kafka-connect-aws-s3/src/main/scala/io/lenses/streamreactor/connect/aws/s3/formats/CsvReaderIteratorAdaptor.scala

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2020 Lenses.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.lenses.streamreactor.connect.aws.s3.formats
+
+import au.com.bytecode.opencsv.CSVReader
+
+class CsvReaderIteratorAdaptor(csvReader: CSVReader) extends Iterator[Array[String]] {
+
+  private var currentRecord: Option[Array[String]] = None
+  private var nextRecord: Option[Array[String]] = Option(csvReader.readNext())
```

Comment:
What happens here if there's nothing left to read. Does `csvReader.readNext` throw an exception? if so this code is not handling it.

Reply:
If there's nothing left to read it will return null, which is why it's wrapped in Option()

It does throw an IOException if (and I quote from the jdoc) "bad things happen during the read".

I was of the opinion that if that happens it's going to error out anyway, but perhaps we want to catch it and throw our own exception?